torch/_subclasses/fake_tensor.py

```diff
@@ -965,6 +977,163 @@ def should_allow_numbers_as_tensors(func: OpOverload):
     )
 
 
+_DISPATCH_CACHE = {}
+
+
+class _Boolean(NamedTuple):
+    value: bool
+
+
+_True = _Boolean(True)
+_False = _Boolean(False)
+
+
+class FakeTensorMetadata(NamedTuple):
+    size: Tuple[int, ...]
+    stride: Tuple[int, ...]
+    dtype: torch.dtype
+    device: torch.device
+    requires_grad: bool
```

Comment:
Do we need to cache on `requires_grad`? I think at this point it should be gone? cc: @eellison

Reply:
In theory there's nothing stopping an operator from looking at this property, I think, so maybe safer to include. Are we also caching non aten/prim ops?